common/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+pub mod types;
+
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use tokio::sync::mpsc;
+use types::*;
+
+/// TODO: Provide error types.
+///
+/// Note: This trait is quite subject to change.
+#[async_trait]
+pub trait AuthorizedNetwork {
+    /// Joins the network with an authorized identity.
+    async fn new(
+        public_key: PublicKey,
+        private_key: PrivateKey,
+        known_ids: Vec<PublicKey>,
+        known_peers: Vec<std::net::Ipv4Addr>,
```

Comment:
is it enough to specify address of peer from only IPv4 Address?
at least, port seems to be needed.

Reply:
As he said, we can insert a struct like `struct Peer {...}` as an additional abstraction.